### ISSUE TITLE
Get rid of Java in favor of Kotlin

### ISSUE THREAD

**Issue by artem-zinnatullin** [artem-zinnatullin/qualitymatters#177]

**Comment by jug6ernaut** [comment]
If this project is meant to showcase best practices for quality Android development i think it would be silly to replace the language that the vast majority of Android developers are using( not to mention is officially supported ). At least bases on my vastly uniformed understanding of current kotlin utilization levels now i believe it would vastly diminish the value of the project.

I am however fully in favor of having portions or even full sections being in kotlin.

**Comment by lenguyenthanh** [comment]
I agree with @artem-zinnatullin, using Kotlin is one of the best practice for Android developers now. But we should keep Java as a separate branch. 

I'd like to help you on this issue, Artem. Please tell me if I can help you anything.

**Comment by artem-zinnatullin** [comment]
Well, I definitely won't be able to support 2 versions of the app in different languages…

>At least bases on my vastly uniformed understanding of current kotlin utilization levels now i believe it would vastly diminish the value of the project.

http://artemzin.com/blog/state-of-kotlin-as-main-language-for-android-project/

@lenguyenthanh thanks for suggesting your help, I'll ping you if something will be required besides code review!

**Comment by jug6ernaut** [comment]
I've read your article :). I meant utilization within the Android development community, not the viability of the language.

**Comment by artem-zinnatullin** [comment]
Well, quality of the code is part of quality matters :)

**Comment by lenguyenthanh** [comment]
@artem-zinnatullin always ready to help :). It's hard to convince people here to use Kotlin :(.

**Comment by danielgomezrico** [comment]
@lenguyenthanh you can add kotlin by parts, I mean, kotlin is 100% interoperable with java, so you can have both files inside same project and compile them, use java from kotlin and kotlin from java, and have a low footprint inside the project because of Kotlin itself

**Comment by serj-lotutovici** [comment]
You can also enable kotlin for functional and integration tests and then move to other parts of the app.

**Comment by adamint** [comment]
Hey, say this was still open. Is this still wanted?

**Comment by artem-zinnatullin** [comment]
Yes
